### PR TITLE
Version page should show build status + not being able to pick a non-built doc

### PR DESCRIPTION
**Pull request by dojutsu-user**
Completes feature request 1 and 3 of issue #1253 

Feature Request 1
screenshot from 2019-01-24 23-14-54

Feature Request 3
screenshot from 2019-01-24 23-15-12

### CODE REVIEW

File: readthedocs/templates/core/project_details.html [hunk 1]
```diff
@@ -41,6 +41,15 @@ <h3>{% trans "Versions" %}</h3>
           {% if request.user|is_admin:project %}
             <ul class="module-item-menu">
               <li><a href="{% url "project_version_detail" project.slug version.slug %}">{% trans "Edit" %}</a></li>
+              <li>
+                <span class="right quiet" style="margin:10px 5px 0px 0px;">
+                  {% if version.built %}
```
Comment (ericholscher) [inline readthedocs/templates/core/project_details.html]: This isn't checking if the build is passing, but only that it has ever passed. We should properly check the last build for this version to get this data.

Comment (dojutsu-user) [reply readthedocs/templates/core/project_details.html]: @ericholscher 
I have implemented a custom template tag for this purpose.

File: readthedocs/templates/core/project_details.html [hunk 2]
```diff
@@ -41,6 +42,16 @@ <h3>{% trans "Versions" %}</h3>
           {% if request.user|is_admin:project %}
             <ul class="module-item-menu">
               <li><a href="{% url "project_version_detail" project.slug version.slug %}">{% trans "Edit" %}</a></li>
+              <li>
+                <span class="right quiet" style="margin:10px 5px 0px 0px;">
```
Comment (stsewd) [inline readthedocs/templates/core/project_details.html]: We shouldn't have inline styles

File: readthedocs/builds/templatetags/builds_tags.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+@register.simple_tag
+def is_latest_built_success(version):
+    """
+    Checks the build status of the passed version.
+
+    Returns true if the latest build for the ``version``
+    is success else returns false. Returns None if no
+    build is found for the given version.
+    """
+    res = version.builds.all().order_by('-date')
```
Comment (stsewd) [inline readthedocs/builds/templatetags/builds_tags.py]: We can just fetch the `success` column from here, not the whole object

**Comment by dojutsu-user** [comment]
@stsewd 
Thank you.
I have pushed the changes as suggested by you.

File: readthedocs/builds/templatetags/builds_tags.py (hunk 3 above)
Comment (dojutsu-user) [reply readthedocs/builds/templatetags/builds_tags.py]: Thanks.
I have pushed the changes.

**Comment by dojutsu-user** [comment]
@stsewd @humitos 
Thank you for guiding me on this.
I have been learning constantly from you two.

File: readthedocs/templates/core/project_details.html [hunk 4]
```diff
@@ -41,6 +42,16 @@ <h3>{% trans "Versions" %}</h3>
           {% if request.user|is_admin:project %}
             <ul class="module-item-menu">
               <li><a href="{% url "project_version_detail" project.slug version.slug %}">{% trans "Edit" %}</a></li>
+              <li>
+                <span class="right quiet status-middle">
+                  {% is_latest_built_success version as is_success %}
```
Comment (ericholscher) [inline readthedocs/templates/core/project_details.html]: This should use this model method: https://github.com/rtfd/readthedocs.org/blob/master/readthedocs/projects/models.py#L882 -- it's likely much more efficient.

**Comment by dojutsu-user** [comment]
@agjohnson 
I have updated the PR with the improvements.
It now looks something like this -
Screenshot from 2019-08-04 13-03-34

File: readthedocs/templates/core/project_details.html (hunk 4 above)
Comment (dojutsu-user) [reply readthedocs/templates/core/project_details.html]: `get_latest_build` returns the latest built of the project -- irrespective of the version.
We need the builds to be of specific version here.

**Pull request closed without merging**